File: src/main/java/alfheim/common/item/ItemStoryToken.java
package alfheim.common.item;

import java.io.IOException;
import java.util.List;

import org.apache.logging.log4j.status.StatusConsoleListener;

import alexsocol.asjlib.ASJUtilities;
import alfheim.AlfheimCore;
import alfheim.api.ModInfo;
import alfheim.common.core.util.AlfheimConfig;
import cpw.mods.fml.common.Loader;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.util.IIcon;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.StatCollector;
import vazkii.botania.common.core.helper.ItemNBTHelper;

public class ItemStoryToken extends Item {
	
	public static final String TAG_STORY = "story";
	public IIcon[] icons = new IIcon[2];
	
	public ItemStoryToken() {
		setCreativeTab(AlfheimCore.alfheimTab);
		setHasSubtypes(true);
		setMaxStackSize(1);
		setUnlocalizedName("StoryToken");
	}
	
	@Override
	public void registerIcons(IIconRegister reg) {
		icons[0] = reg.registerIcon(ModInfo.MODID + ":StoryToken");
		icons[1] = reg.registerIcon(ModInfo.MODID + ":StoryTokenSigned");
	}
	
	@Override
	public IIcon getIconFromDamage(int meta) {
		return icons[meta == 1 ? 1 : 0];
	}
	
	@Override
	public void addInformation(ItemStack stack, EntityPlayer player, List info, boolean extra) {
		if (stack.getItemDamage() == 1) {
			if (!ItemNBTHelper.verifyExistance(stack, TAG_STORY)) ItemNBTHelper.setInt(stack, TAG_STORY, player.getRNG().nextInt(AlfheimConfig.storyLines) + 1);
			info.add(StatCollector.translateToLocal(String.format("item.StoryToken.story.%d",
					 ItemNBTHelper.getInt(stack, TAG_STORY, 0))));
		}
	}
}

File: src/main/java/alfheim/common/item/material/ItemStoryToken.kt
package alfheim.common.item.material

import alexsocol.asjlib.meta
import alfheim.client.core.helper.IconHelper
import alfheim.common.core.handler.AlfheimConfigHandler
import alfheim.common.core.util.AlfheimTab
import alfheim.common.item.ItemMod
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.entity.Entity
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.*
import net.minecraft.world.World
import vazkii.botania.api.recipe.IElvenItem
import vazkii.botania.common.core.helper.ItemNBTHelper

class ItemStoryToken: ItemMod("StoryToken"), IElvenItem {
	
	lateinit var icons: Array<IIcon>
	
	init {
		creativeTab = AlfheimTab
		setHasSubtypes(true)
		setMaxStackSize(1)
	}
	
	override fun registerIcons(reg: IIconRegister) {
		icons = Array(2) { IconHelper.forItem(reg, this, it) }
	}
	
	override fun getIconFromDamage(meta: Int): IIcon {
		return icons[if (meta == 1) 1 else 0]
	}
	
	override fun onUpdate(stack: ItemStack, world: World, entity: Entity, slotID: Int, inHand: Boolean) {
		if (world.isRemote) return
		
		if (stack.meta == 1 && !ItemNBTHelper.verifyExistance(stack, TAG_STORY))
			ItemNBTHelper.setInt(stack, TAG_STORY, world.rand.nextInt(AlfheimConfigHandler.storyLines) + 1)
	}
	
	override fun addInformation(stack: ItemStack, player: EntityPlayer, info: MutableList<Any?>, adv: Boolean) {
		if (stack.meta == 1) {
			info.add(StatCollector.translateToLocal("item.StoryToken.story.${ItemNBTHelper.getInt(stack, TAG_STORY, 0)}"))
		}
	}
	
	override fun isElvenItem(stack: ItemStack) = stack.meta == 1
	
	companion object {
		
		val TAG_STORY = "story"
	}
}
